client_test.go

```diff
@@ -68,6 +68,59 @@ func newLocalListenerUDP(t *testing.T) (*net.UDPConn, *net.UDPAddr) {
 	return ln, udpAddr
 }
 
+func TestTiming(t *testing.T) {
+	addr, ln := newLocalListenerTCP(t)
+	defer ln.Close()
```

Comment:
Is it expected that you're only testing on TCP? If so, you might want to change the test name to reflect that.

Reply:
I figure with UDP there is no guarantee that these will come back in the same order, so it would make it harder to validate the output. Same reason I changed s from a map to a list of structs.